### PR TITLE
refactor(autoware_pointcloud_preprocessor): rework voxel grid downsample filter parameters

### PR DESCRIPTION
**Pull request by vividf**
## Description
This PR includes the following changes
1. Remove the default parameters for the voxel grid downsample filter node. 
2. Add parameter, launch file, and parameters schema for the node. 
3. Change the name `nodelet` to `node`.

A big thank you to @Ariiees for the PRs (https://github.com/autowarefoundation/autoware.universe/pull/8297 and https://github.com/autowarefoundation/autoware.universe/pull/7422), where we worked together on the entire pointcloud preprocessor parameters, launch file, and schema.

## Related links

**Parent Issue:**

- Link

## How was this PR tested?
`ros2 launch autoware_pointcloud_preprocessor voxel_grid_downsample_filter_node.launch.xml`
## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: sensing/autoware_pointcloud_preprocessor/src/downsample_filter/voxel_grid_downsample_filter_node.cpp [hunk 1]
```diff
@@ -66,9 +66,9 @@ VoxelGridDownsampleFilterComponent::VoxelGridDownsampleFilterComponent(
 {
   // set initial parameters
   {
-    voxel_size_x_ = static_cast<float>(declare_parameter("voxel_size_x", 0.3));
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/src/downsample_filter/voxel_grid_downsample_filter_node.cpp:69]: In the past, declaring parameters as `float` was not possible.
Are you sure it is ok now?

For reference:
https://github.com/ros2/rclcpp/blob/e6b6faf152ae3c758f73a35ad1385bfd987f346f/rclcpp/include/rclcpp/parameter_value.hpp#L35

**Review by knzo25 — CHANGES_REQUESTED** [review]
I left a question regarding the internals of the parameters in ROS

File: sensing/autoware_pointcloud_preprocessor/src/downsample_filter/voxel_grid_downsample_filter_node.cpp (hunk 1 above)
Comment (knzo25) [reply sensing/autoware_pointcloud_preprocessor/src/downsample_filter/voxel_grid_downsample_filter_node.cpp:69]: update: they did not change the old structure but added other stl tricks to add compatibility

**Review by knzo25 — APPROVED** [review]
LGTM

File: sensing/autoware_pointcloud_preprocessor/src/downsample_filter/voxel_grid_downsample_filter_node.cpp (hunk 1 above)
Comment (vividf) [reply sensing/autoware_pointcloud_preprocessor/src/downsample_filter/voxel_grid_downsample_filter_node.cpp:69]: Thanks!!

**Pull request merged**